### ISSUE TITLE
Unable to resolve grails.convertors.JSON

### ISSUE THREAD

**Issue by aminroosta** [dansomething/coc-groovy#20]
I have a big gradle project, but many of the internal modules as well as grails modules are not resolved, therefore the plugin doesn't work.

How could i debug this? :)

**Comment by aminroosta** [comment]
<summary>
And after two minutes I get these errors, in <code>:messages</code>
<details>
<pre>
<code>
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1597
[coc.nvim]: UnhandledRejection: Cannot call write after a stream was destroyed
[coc.nvim]: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
[coc.nvim]:     at doWrite (_stream_writable.js:399:19)
[coc.nvim]:     at writeOrBuffer (_stream_writable.js:387:5)
[coc.nvim]:     at Socket.Writable.write (_stream_writable.js:318:11)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3107
[coc.nvim]:     at new Promise (<anonymous>)
[coc.nvim]:     at IN.write (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3028)
[coc.nvim]:     at z0.doWrite (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1698)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1597
[coc.nvim]: UnhandledRejection: Cannot call write after a stream was destroyed
[coc.nvim]: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
[coc.nvim]:     at doWrite (_stream_writable.js:399:19)
[coc.nvim]:     at writeOrBuffer (_stream_writable.js:387:5)
[coc.nvim]:     at Socket.Writable.write (_stream_writable.js:318:11)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3107
[coc.nvim]:     at new Promise (<anonymous>)
[coc.nvim]:     at IN.write (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3028)
[coc.nvim]:     at z0.doWrite (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1698)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1597
[coc.nvim]: UnhandledRejection: Cannot call write after a stream was destroyed
[coc.nvim]: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
[coc.nvim]:     at doWrite (_stream_writable.js:399:19)
[coc.nvim]:     at writeOrBuffer (_stream_writable.js:387:5)
[coc.nvim]:     at Socket.Writable.write (_stream_writable.js:318:11)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3107
[coc.nvim]:     at new Promise (<anonymous>)
[coc.nvim]:     at IN.write (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3028)
[coc.nvim]:     at z0.doWrite (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1698)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1597
[coc.nvim]: UnhandledRejection: Cannot call write after a stream was destroyed
[coc.nvim]: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
[coc.nvim]:     at doWrite (_stream_writable.js:399:19)
[coc.nvim]:     at writeOrBuffer (_stream_writable.js:387:5)
[coc.nvim]:     at Socket.Writable.write (_stream_writable.js:318:11)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3107
[coc.nvim]:     at new Promise (<anonymous>)
[coc.nvim]:     at IN.write (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:33:3028)
[coc.nvim]:     at z0.doWrite (/Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1698)
[coc.nvim]:     at /Users/amin.roosta/.vim/plugged/coc.nvim/build/index.js:34:1597
[coc.nvim]: UnhandledRejection: Cannot call write after a stream was destroyed
[coc.nvim]: Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
[coc.nvim]:     at doWrite (_stream_writable.js:399:19)
</code>
</pre>
</details>
</summary>

**Comment by dansomething** [comment]
Was this the only relevant output in `:messages`? It seems to point to an error occurring in coc.nvim.

As far as debugging goes:
1) Try increasing the log output of coc.nvim by using the `NVIM_COC_LOG_LEVEL` [environment variable](https://github.com/neoclide/coc.nvim/wiki/Environment-variables).
1) View the coc logs using `:CocOpenLog`
1) Take a look at the logs available from `:CocCommand workspace.showOutput`
1) Try [debugging coc.nvim or the coc-groovy extension](https://github.com/neoclide/coc.nvim/wiki/Debug-coc.nvim#use-debugger-of-nodejs).

Let me know if you capture any more useful logs or other information to help troubleshoot.

**Comment by aminroosta** [comment]
I got this error from `CocCommand workspace.showOutput`.
I'm using grails 3.3.11 which uses groovy 2.4.17  internally.
There seems to be [(related?) bugs](https://issues.apache.org/jira/browse/GROOVY-8156) on groovy 2.4.x, however unfortunately I can't upgrade grails yet.

```
Unexpected exception in language server when compiling Groovy.
BUG! exception in phase 'canonicalization' in source unit '/Users/amin/projects/backend/grails-app/jobs/company/ReviewAnalysisJob.groovy' ClassNode#getTypeClass for company.RetryOnException is called before the type class is set 
	at org.codehaus.groovy.ast.ClassNode.getTypeClass(ClassNode.java:1387)
	at org.codehaus.groovy.ast.ClassNode.getTypeClass(ClassNode.java:1381)
	at org.codehaus.groovy.transform.trait.Traits.findHelpers(Traits.java:149)
	at org.codehaus.groovy.transform.trait.TraitComposer.doExtendTraits(TraitComposer.java:110)
	at org.codehaus.groovy.control.CompilationUnit$5.call(CompilationUnit.java:235)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1084)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:640)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:618)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:595)
	at net.prominic.groovyls.GroovyServices.compile(GroovyServices.java:448)
	at net.prominic.groovyls.GroovyServices.compileAndVisitAST(GroovyServices.java:431)
	at net.prominic.groovyls.GroovyServices.didOpen(GroovyServices.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


```

**Comment by dansomething** [comment]
Good catch. It does appear to be an issue with the Groovy language server. Strangely the bug says it was fixed 2.4.11 and the Groovy Language server appears to be using 2.5.14.
